src/Shared/LayoutRenderers/AspNetRequestValueLayoutRenderer.cs

```diff
@@ -98,11 +97,24 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
             {
                 value = LookupCookieValue(Cookie, httpRequest);
             }
-#if !ASP_NET_CORE
+#if !ASP_NET_CORE || ASP_NET_CORE3
             else if (ServerVariable != null)
             {
+    #if !ASP_NET_CORE
                 value = httpRequest.ServerVariables?.Count > 0 ?
                     httpRequest.ServerVariables[ServerVariable] : null;
+    #elif ASP_NET_CORE3
+                var features = HttpContextAccessor.HttpContext.TryGetFeatureCollection();
```

Comment:
Just like `LookupCookieValue` can this be extracted into method called `LookupServerVariable` ?

Reply:
Complete